### PR TITLE
GEOMESA-2150 Users can configure StreamDataStore with the name of an …

### PR DESCRIPTION
**Pull request by anthonyccri**
…SFT and name of a converter

* Added options for configuring StreamDataStore
* Added no-timeout option

Signed-off-by: Anthony Fox <anthonyfox@ccri.com>

### CODE REVIEW

File: geomesa-stream/geomesa-stream-datastore/src/main/scala/org/locationtech/geomesa/stream/datastore/StreamDataStore.scala [hunk 1]
```diff
@@ -145,28 +156,78 @@ class StreamFeatureStore(entry: ContentEntry,
...
 }
 
 class StreamDataStoreFactory extends DataStoreFactorySpi {
 
   import StreamDataStoreParams._
+  private val log = LoggerFactory.getLogger(classOf[StreamDataStoreFactory])
 
   override def createDataStore(params: ju.Map[String, java.io.Serializable]): DataStore = {
-    val confString = StreamDatastoreConfig.lookup(params)
+    // TODO: note, if user is requesting a SDS via a inline config, it'll get a default name and namespace of ":"
```
Comment (jnh5y) [inline geomesa-stream/geomesa-stream-datastore/src/main/scala/org/locationtech/geomesa/stream/datastore/StreamDataStore.scala]: @elahrvivaz do we have a standard way of handling this?

Comment (elahrvivaz) [reply geomesa-stream/geomesa-stream-datastore/src/main/scala/org/locationtech/geomesa/stream/datastore/StreamDataStore.scala]: we have a standard way of handling datastore namespaces, but i think the stream store is actually using the namespace, which we don't do anywhere else.

File: geomesa-stream/geomesa-stream-datastore/src/main/scala/org/locationtech/geomesa/stream/datastore/StreamDataStore.scala [hunk 2]
```diff
@@ -145,28 +156,78 @@ class StreamFeatureStore(entry: ContentEntry,
...
+}
+
+object StreamDataStoreFactory {
+  val stores: ju.Map[String, StreamDataStore] = Collections.synchronizedMap(Maps.newHashMap[String, StreamDataStore]())
 }
 
 class StreamDataStoreFactory extends DataStoreFactorySpi {
 
   import StreamDataStoreParams._
+  private val log = LoggerFactory.getLogger(classOf[StreamDataStoreFactory])
```
Comment (jahhulbert-ccri) [inline geomesa-stream/geomesa-stream-datastore/src/main/scala/org/locationtech/geomesa/stream/datastore/StreamDataStore.scala]: what is this java?

File: geomesa-stream/geomesa-stream-datastore/src/main/scala/org/locationtech/geomesa/stream/datastore/StreamDataStore.scala [hunk 3]
```diff
@@ -46,27 +54,27 @@ case class FeatureHolder(sf: SimpleFeature, env: Envelope) {
...
-      .newBuilder()
-      .expireAfterWrite(timeout, TimeUnit.SECONDS)
-      .removalListener(
-        new RemovalListener[String, FeatureHolder] {
-          override def onRemoval(k: String, v: FeatureHolder, removalCause: RemovalCause): Unit = {
-            qt.remove(v.env, v.sf)
-          }
+  val cb = {
+    val builder = Caffeine.newBuilder()
+    if(timeout > 0) builder.expireAfterWrite(timeout, TimeUnit.SECONDS)
```
Comment (jahhulbert-ccri) [inline geomesa-stream/geomesa-stream-datastore/src/main/scala/org/locationtech/geomesa/stream/datastore/StreamDataStore.scala]: [picky] space after if and i like curlies {}

File: geomesa-stream/geomesa-stream-datastore/src/main/scala/org/locationtech/geomesa/stream/datastore/StreamDataStore.scala [hunk 4]
```diff
@@ -46,27 +54,27 @@ case class FeatureHolder(sf: SimpleFeature, env: Envelope) {
   }
 }
 
-class StreamDataStore(source: SimpleFeatureStreamSource, timeout: Int) extends ContentDataStore {
+class StreamDataStore(source: SimpleFeatureStreamSource, timeout: Int, ns: Option[String]) extends ContentDataStore {
```
Comment (jahhulbert-ccri) [inline geomesa-stream/geomesa-stream-datastore/src/main/scala/org/locationtech/geomesa/stream/datastore/StreamDataStore.scala]: i think preferably timeout would be either: 
1. defined as long as millis 
2. require a timeunit
3. be a string and use the time unit parsing code (e.g. "3 min", "6 hours")

Comment (elahrvivaz) [reply geomesa-stream/geomesa-stream-datastore/src/main/scala/org/locationtech/geomesa/stream/datastore/StreamDataStore.scala]: I'll add one

File: geomesa-stream/geomesa-stream-datastore/src/main/scala/org/locationtech/geomesa/stream/datastore/StreamDataStore.scala (hunk 3 above)
Comment (elahrvivaz) [reply geomesa-stream/geomesa-stream-datastore/src/main/scala/org/locationtech/geomesa/stream/datastore/StreamDataStore.scala]: fixed

File: geomesa-stream/geomesa-stream-datastore/src/main/scala/org/locationtech/geomesa/stream/datastore/StreamDataStore.scala (hunk 4 above)
Comment (elahrvivaz) [reply geomesa-stream/geomesa-stream-datastore/src/main/scala/org/locationtech/geomesa/stream/datastore/StreamDataStore.scala]: added as a TODO: https://geomesa.atlassian.net/browse/GEOMESA-2161

**Pull request merged**